mwclient/client.py

```diff
@@ -1485,14 +1797,14 @@ def expandtemplates(self, text, title=None, generatexml=False):
         else:
             return result['expandtemplates']['*']
 
-    def ask(self, query, title=None):
+    def ask(self, query: str, title: Optional[str] = None) -> Iterable[Dict[str, Any]]:
```

Comment:
Shouldn't we be using `Generator` here, not `Iterable`?

Reply:
We could use `Generator[Dict[str, Any], None, None]`, but according to the docs (https://docs.python.org/3/library/typing.html#typing.Generator) it's fine to use `Iterator[Dict[str, Any]]` instead. I think this is a bit more readable as it is, but am also fine with changing it.

Reply:
oh, sure, if the docs say it's fine I guess it's fine.